std/container/rbtree.d

```diff
@@ -801,9 +800,7 @@ if (is(typeof(binaryFun!less(T.init, T.init))))
 
     static private Node allocate(Elem v)
     {
-        auto result = allocate();
```

Comment:
As far as I can see, the idea behind `allocate()` is to abstract the allocation mechanism. However, in practice it doesn't offer any flexibility, so we can even remove it. But that's better left for another PR.